client/package-lock.json

```diff
@@ -0,0 +1,5879 @@
+{
```

Comment:
What consumes this file? I don't think it should be included in this repo. I expect it to change often and unnecessarily bloat PR's. Same with package.json (to a lesser degree). Please add it to the .gitignore.

Reply:
This is consumed by the linting CI task. Like I said in my other project, `package-lock.json` stores _specific_ versions of packages that are installed, which is used by commands like `npm ci` to make sure that what is installed on another computer matches _exactly_ what is installed on your own, so that there are no mysterious errors. `package.json` only specifies approximate versions of packages, which means there is a chance that a package could get silently upgraded and break everything.

And it won't change often. Since I can't preprocess the frontend code, I can't use any Node.js dependencies at runtime anyway. This file and `package.json` are just for ESLint and Stylelint.

Reply:
My understanding was that linting would happen in the Github Workflow and potentially in a dev's code editor if they enabled it through an extension. We shouldn't be requiring everyone to download NodeJS just to enable linting. The burden of work is unnecessary and disproportionately affects the other team members. Both this and package.json should be removed.

Reply:
If you don't want to put NodeJS on your computer (which is not hard, I wrote instructions on how to do it already), then you can just upload the PR and GitHub will lint it for you. If I don't have `package.json` in here, then GitHub cannot perform any linting.

Reply:
Fair enough. If it's needed for the github workflow, then it should be included here. Note, though, that teammates should not feel compelled to lint locally since Github will do it for them.